buildSrc/src/main/kotlin/org/sonarsource/kotlin/buildsrc/tasks/CreateRuleStubsTask.kt

```diff
@@ -48,47 +42,46 @@ abstract class CreateRuleStubsTask : DefaultTask() {
 
         val finishedTasks = listOf(
-            "Create Check Class" to createCheckClass(checkClassName),
-            "Create Test Class" to createTestClass(checkClassName),
-            "Create Sample File" to createSampleFile(checkClassName),
-            "Add Rule to KotlinCheckList" to addRuleToChecksListFile(checkClassName),
+            "Create Check Class" to templates.createCheckClass(checkClassName),
+            "Create Test Class" to templates.createTestClass(checkClassName),
+            "Create Sample File" to templates.createSampleFile(checkClassName),
+            "Add Rule to KotlinCheckList" to templates.addRuleToChecksListFile(checkClassName),
         )
 
         logger.info("--------- Done ---------")
         finishedTasks.forEach { logger.info("${it.first}: ${if (it.second) "successful" else "FAILED"}") }
     }
 
-    private fun createCheckClass(checkClassName: String) =
-        createNewFile(
-            checksDir.resolve("${checkClassName}.kt"),
-            generateCheckClass(ruleKey, checkClassName, message?.let { """private const val MESSAGE = "$it"""" })
-        )
+    private fun RuleStubTemplates.createCheckClass(checkClassName: String) = createNewFile(
+        checksDir.resolve("${checkClassName}.kt"),
+        generateCheckClass(ruleKey, checkClassName, message?.let { """private const val MESSAGE = "$it"""" })
+    )
 
-    private fun createTestClass(checkClassName: String): Boolean {
+    private fun RuleStubTemplates.createTestClass(checkClassName: String): Boolean {
         val testClassName = "${checkClassName}Test"
         return createNewFile(testsDir.resolve("${testClassName}.kt"), generateTestClass(testClassName, checkClassName))
     }
 
-    private fun createSampleFile(checkClassName: String) =
-        createNewFile(samplesDir.resolve("${checkClassName}Sample.kt"), generateCheckFile("${checkClassName}Sample", ruleKey))
+    private fun RuleStubTemplates.createSampleFile(checkClassName: String) =
+        createNewFile(samplesDir.resolve("${checkClassName}Sample.$sampleFileExt"), generateCheckFile("${checkClassName}Sample", ruleKey))
 
-    private fun createNewFile(targetFile: Path, content: String) =
-        if (targetFile.notExists()) {
-            targetFile.createFile()
-            targetFile.writeText(content)
-            true
-        } else {
-            logger.warn("WARNING: File '$targetFile' exists. Not creating a new one.")
-            false
-        }
+    private fun createNewFile(targetFile: Path, content: String) = if (targetFile.notExists()) {
+        targetFile.createFile()
+        targetFile.writeText(content)
+        true
+    } else {
+        logger.warn("WARNING: File '$targetFile' exists. Not creating a new one.")
+        false
+    }
 
-    private fun addRuleToChecksListFile(checkClassName: String): Boolean {
+    private fun RuleStubTemplates.addRuleToChecksListFile(checkClassName: String): Boolean {
```

Comment:
There are a lot of functions that have become extension functions. While extension functions can be a valuable tool, in this case we can simply move the functions into the corresponding type, making the code easier to understand. Perhaps then we should also rethink the "Template" naming and make it a "Factory"/"Builder"/"Creator"/"Generator" or so instead.